examples/multi_cartpole/controllers/supervisor_manager/supervisor_controller.py

```diff
@@ -0,0 +1,283 @@
+            exit()
+
+        self.handle_emitter(action)
+
+        return (
+            self.get_observations(),
+            self.get_reward(action),
+            self.is_done(),
+            self.get_info(),
+        )
+
+    def handle_emitter(self, actions):
+        """
+        Emits actions to the robots through robot specific emitter channels
+        """
+        for i, action in enumerate(actions):
+            message = str(action).encode("utf-8")
+            self.communication[i]['emitter'].send(message)
+
+    def handle_receiver(self):
+        """
+        Receives actions sent by the robots in the environment through robot specific receiver channels
+        :return messages: list of messages sent by the robots
+        :rtype: list(int)
+        """
+        messages = []
+        for com in self.communication:
+            receiver = com['receiver']
+            if receiver.getQueueLength() > 0:
+                messages.append(receiver.getData().decode("utf-8"))
+                receiver.nextPacket()
+            else:
+                messages.append(None)
+
+        return messages
+    
+    def get_observations(self):
+        """
+        This get_observation implementation builds the required observations for the MultiCartPole problem.
+        All values apart from pole angle are gathered here from the robots and poleEndpoint objects.
+        The pole angle value is taken from the messages sent by the robots.
+        All values are normalized appropriately to [-1, 1], according to their original ranges.
+
+        :return: Observation:[[cartPosition0, cartVelocity0, poleAngle0, poleTipVelocity0],
+                              [cartPosition1, cartVelocity1, poleAngle1, poleTipVelocity1], ...
+                              [cartPosition9, cartVelocity9, poleAngle9, poleTipVelocity9]]
+        :rtype: list(list(list(float), list(float), float, float))   
+        """
+        # Position on z axis
+        cartPosition = [normalizeToRange(self.robot[i].getPosition()[0], -0.4, 0.4, -1.0, 1.0) for i in range(self.num_robots)]
+
+        # Linear velocity on z axis
+        cartVelocity = [normalizeToRange(self.robot[i].getVelocity()[0], -0.2, 0.2, -1.0, 1.0, clip=True) for i in range(self.num_robots)]
+
+        self.messageReceived = self.handle_receiver()
+
+        poleAngle = [None for _ in range(self.num_robots)]
+
+        for i, message in enumerate(self.messageReceived):    
+            if message is not None:
+                poleAngle[i] = normalizeToRange(message, -0.23, 0.23, -1.0, 1.0, clip=True)
+            else:
+                # method is called before messageReceived is initialized
+                poleAngle = [0.0 for _ in range(self.num_robots)]
+
+        # Angular velocity x of endpoint
+        endpointVelocity = [normalizeToRange(self.poleEndpoint[i].getVelocity()[3], -1.5, 1.5, -1.0, 1.0, clip=True) for i in range(self.num_robots)]
```

Comment:
This might be the reason why it's taking too long to converge / not converging.
```suggestion
        endpointVelocity = [normalizeToRange(self.poleEndpoint[i].getVelocity()[4], -1.5, 1.5, -1.0, 1.0, clip=True) for i in range(self.num_robots)]
```